### ISSUE TITLE
Load info from Salesforce

### ISSUE THREAD

**Issue by luccisa** [katzwebservices/Gravity-Forms-Salesforce#286]
I posted this in wordpress before I realized it should be posted here.
 
We would like to have drop down menus that pull the info / options from SalesForce rather than setting them manually.

For example, we want our staff to be able to pick from various jobs to report on through the website, but these jobs change frequently / daily in Salesforce. Rather than having to create new and matching jobs in both Saleforce and Gravity forms, is there a way to have the jobs change in Salesforce only and then automatically have those show up in the drop down in the Gravity form?

**Comment by datapharmer** [comment]
I agree that this would be very useful. There is a partial manual workaround for this (see thread here for example: https://github.com/katzwebservices/Gravity-Forms-Salesforce/issues/156 ) 

but it would be very nice and useful to be able to specify a salesforce object (perhaps via css, admin label or parameter) and filters within gravity forms instead of needing to use pre_render.

**Comment by datapharmer** [comment]
Here is a rough/ugly example (add to functions.php, set case name to be the css name attached to the form item and modify the query to meet your needs)
`add_filter("gform_pre_render_yourformnumberhere", "pre_render_salesforce");


function pre_render_salesforce( $form ){
	$api		= GFSalesforce::get_api();

	foreach($form['fields'] as &$field){
	$cssClass = $field->cssClass;
		switch ($cssClass) {
		case "ProductsFromFamily":
			$query 		= 'SELECT Id, Name, UnitPrice FROM PriceBookEntry  Where Product2.Family LIKE \'Product Family Name Goes Here\' AND (Not Product2.Name Like \'%Excluded Product Name Goes here%\') Order By Name DESC'; //Alter as need
			$results 	= $api->query($query);
			$items = array();
			foreach($results->records as $result) {
				$SFstr = preg_replace('{sf:}', '', $result->any);
				$SFxml = '<?xml version="1.0" encoding="UTF-8"?><obj>'.$SFstr.'</obj>';
				$SFobject = simplexml_load_string($SFxml);
				$any 	= strip_tags(  $result->any );
				$id 	= $result->Id[0];
				$Name = $SFobject->Name;
				$price = $SFobject->UnitPrice;
				$items[] 	= array( "text" => $Name, "value" => $id, "price"=> $price );
			}
			$field["choices"] = $items;
			break;
		case "OtherProducts":
			$query 		= 'SELECT Id,Name FROM PriceBookEntry Where Product2.Family = \'NameOfProductFamily\' order by Product2.Name'; //Alter as need
			$results 	= $api->query($query);
			$items = array();
			foreach($results->records as $result) {
				$any 		= strip_tags(  $result->any );
				$id = $result->Id[0];
				//price and quantity are hardcoded for this example, but this is optional. see case above.
				$items[] 	= array( "text" => $any, "value" => $id, "price"=>"5.5", "quantity"=>"1" );
			}
			$field["choices"] = $items;
			break;
		}		
	}

	return $form;
}`
